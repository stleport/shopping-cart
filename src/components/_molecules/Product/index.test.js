import React from 'react';
import {
  shallow
} from 'enzyme';
import renderer from 'react-test-renderer';
import {
  ProductCard
} from '.';

function ProductCardSetup() {
  const props = {
    product: {
      productId: 1,
      title: 'Sabl√©s',
      inventoryLevel: 'AVAILABLE',
      shortDescription: 'Description',
      images: ['http://example.com/image.jpg']
    },
    cart: {
      items: [{
        productId: 1,
        quantity: 2
      }],
      total: 0
    },
    item: {
      productId: 1
    },
    onAddToCart: jest.fn(),
    onSubQuantity: jest.fn()
  };

  const ProductCardJSX = (
    <ProductCard {...props} />
  );

  const ProductCardComponent = shallow(ProductCardJSX);
  return {
    ProductCardComponent,
    ProductCardJSX,
    props
  };
}

describe('ProductCard', () => {
  it('renders correctly ProductCard component', () => {
    const {
      ProductCardJSX
    } = ProductCardSetup();
    const ProductCardComponent = renderer.create(ProductCardJSX).toJSON();
    expect(ProductCardComponent).toMatchSnapshot();
  });
});
