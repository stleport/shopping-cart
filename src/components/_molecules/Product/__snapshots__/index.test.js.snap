// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly ProductCard component 1`] = `
<div
  className="st-Card"
>
  <div
    className="st-Card__image"
  >
    <img
      alt="Sablés"
      className="image"
      onError={[Function]}
      src="http://example.com/image.jpg"
    />
    <div>
      <div
        className="st-Card__overlay"
      >
        <h3>
          2
           portion
          s
           dans votre Frichti
        </h3>
      </div>
    </div>
  </div>
  <div
    className="st-Card__content"
  >
    <h4>
      Sablés
    </h4>
  </div>
  <div
    className="st-Card__extra"
  >
    <div
      className="st-Card__pricing"
    >
      NaN €
    </div>
    <div
      className="st-Card__actions"
    >
      <button
        className="st-Edit-button__button st-Edit-button__button--dark button disabled"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="minus"
          className="st-Edit-button__icon--white"
          fill="white"
          src="minus.svg"
        />
      </button>
      <button
        className="st-Edit-button__button st-Edit-button__button--light buttontrue"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="plus"
          className="st-Edit-button__icon--dark"
          fill="white"
          src="plus.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
